internal/worker/rundynamic.go

```diff
@@ -148,11 +152,60 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
 	return result, nil
 }
 
+// openStraceDebugLogFile returns the file to be used for debug logging of strace parsing during a
+// dynamic analysis phase, or nil if logging should be disabled or an error occurred while opening it.
+// The file is created with the given filename. It should be closed by the caller.
+// The directory path (within the analysis container) is configured via setting a (string) value
+// for the log.StraceDebugLogDirKey key on the context object passed to this function. If there
+// is no directory configured, logging will be disabled
+func openStraceDebugLogFile(ctx context.Context, name string) *os.File {
+	if !featureflags.StraceDebugLogging.Enabled() {
+		return nil
+	}
+
+	var logDir string
+	if v := ctx.Value(log.StraceDebugLogDirKey); v == nil {
+		return nil
+	} else if dirPath, ok := v.(string); !ok {
+		slog.WarnContext(ctx, "non-string value for log.StraceDebugLogDirKey", "value", v)
+		return nil
+	} else {
+		logDir = dirPath
+	}
+
+	if err := os.MkdirAll(logDir, 0o777); err != nil {
+		slog.WarnContext(ctx, "could not create directory for strace debug log file", "path", logDir, "error", err)
+	}
+
+	logPath := filepath.Join(logDir, name)
+	if logFile, err := os.Create(logPath); err != nil {
+		slog.WarnContext(ctx, "could not create strace debug log file", "path", logPath, "error", err)
+		return nil
+	} else {
+		return logFile
+	}
+}
+
+func straceDebugLogFilename(pkg *pkgmanager.Pkg, phase analysisrun.DynamicPhase) string {
+	return fmt.Sprintf("strace-%s-%s-%s-%s.log", pkg.EcosystemName(), pkg.Name(), pkg.Version(), phase)
```

Comment:
Just a heads up, NPM packages with a "/" in the name are likely going to cause problems with this filename.

Reply:
Nice catch, thanks